tests/async/test_browsertype_connect.py

```diff
@@ -241,3 +243,54 @@ async def handle_request(route: Route) -> None:
     assert await response.json() == {"foo": "bar"}
 
     remote.kill()
+
+@pytest.mark.only_browser("chromium")
+async def test_should_upload_large_file(
+    browser_type: BrowserType,
+    launch_server: Callable[[], RemoteServer],
+    playwright: Playwright,
+    server: Server,
+    tmp_path):
+    remote = launch_server()
+
+    browser = await browser_type.connect(remote.ws_endpoint)
+    context = await browser.new_context()
+    page = await context.new_page()
+
+    await page.goto(server.PREFIX + "/input/fileupload.html")
+    large_file_path = tmp_path / "200MB.zip"
+    data = b'A' * 1024
+    with large_file_path.open("wb") as f:
+        for i in range(0, 200 * 1024 * 1024, len(data)):
+            f.write(data)
+    input = page.locator('input[type="file"]')
+    events = await input.evaluate_handle(
+        """
+        e => {
+            const events = [];
+            e.addEventListener('input', () => events.push('input'));
+            e.addEventListener('change', () => events.push('change'));
+            return events;
+        }
+    """
+    )
+
+    await input.set_input_files(large_file_path)
+    assert await input.evaluate("e => e.files[0].name") == "200MB.zip"
+    assert await events.evaluate("e => e") == ["input", "change"]
+
+    file_upload = asyncio.Future()
+    def handler(request):
+        file_upload.set_result(request)
+        request.finish()
+    server.set_route("/upload", handler)
+
+    await page.click("input[type=submit]")
+    request = await file_upload
+    contents = request.args[b'file1'][0]
+    assert len(contents) == 200 * 1024 * 1024
+    assert contents[:1024] == data
```

Comment:
why is there a 1024 offset?

Reply:
we're just checking the first and last 1024 bytes in the test; assuming those are correct the other 200MB should be good in between 😄

Reply:
gotcha